### PR TITLE
DF-333:Update 'Collection' and 'Held by' filters to use 'OR' rather than the default 'AND' operator

### PR DESCRIPTION
**Pull request by TNA-Allan**
Make collection, heldby fields to use OR operator for multiple filter values

### CODE REVIEW

**Comment by ababic** [comment]
Sorry @TNA-Allan, but I'm not sure that hardcoding the treatment of specific filter aggregations is the way to go (it's quite possible that other views could be introduced in future that want to apply multiple AND filters for the same field). I think it should be up to the Django view to add ":or" to each of the form values instead.

**Comment by ababic** [comment]
@TNA-Allan I would suggest [BaseFilteredSearchView.get_api_filter_aggregations()](https://github.com/nationalarchives/ds-wagtail/blob/develop/etna/search/views.py#L403-L425) as the place for this... perhaps using an `or_filters = []` attribute on the class to control it.

**Comment by TNA-Allan** [comment]
> Sorry @TNA-Allan, but I'm not sure that hardcoding the treatment of specific filter aggregations into the client is the way to go (it's quite possible that other views could be introduced in future that want to apply multiple AND filters for the same field). I think it should be up to the Django view to add ":or" to each of the form values instead.

I'd like to understand what is hardcoding the treatment means
since the code is placed in `def prepare_filter_aggregations` where these fields are treated specifically
using configured `fields_using_or_operator = ["heldBy", "collection"]`
the `":or"` is appended at one place.

Besides this as the method name states, it does `prepare` the filter. Also it takes care of `:` which replaces values containing spaces and not in the `:or` appended.

Can you please give an example of applying multiple AND filters for the same field?

**Comment by ababic** [comment]
@TNA-Allan my advice would be to try and step back from the problem a little and see each component for what it is...

The clients job is to make an API request based on the options passed to it's methods. It's a 'convenience' interface t

**Comment by TNA-Allan** [comment]
> Now, if I were to ask you "Allan, the search view appears to be doing something funny when I use this combination of filters. Can you look into it?", where would you look first?

I'm thinking of fat models, skinny views, although this is not the same  :)

> However unlikely it is that the user intends to do an multiple `AND` search for these particular fields is redundant. It should be apparent from the view code (the first place you should look when someone reports a problem) how the values are being treated and passed to the API client.

Using of AND in this case is meaningless as the data returned with AND is mutually exclusive, therefore 0 records returned when AND applied.

`def prepare_filter_aggregations` is in client will require further treatment due to the `:or` appearing at the end. Should this method qualify to be moved over to the view and processed over here [BaseFilteredSearchView.get_api_filter_aggregations()](https://github.com/nationalarchives/ds-wagtail/blob/develop/etna/search/views.py#L403-L425)

File: etna/ciim/client.py [hunk 1]
```diff
@@ -107,8 +110,24 @@ def prepare_filter_aggregations(items: Optional[list]) -> Optional[str]:
...
+        if sum((item.split(":", 1)[0].count(field) for item in items)) > 1:
+            # append or to the value for each field
+            updated_list_for_or_operator = []
+            for item in filter_prepared_list:
+                if item.split(":", 1)[0] == field:
+                    updated_list_for_or_operator.append(item + ":or")
+                else:
+                    updated_list_for_or_operator.append(item)
+
+            filter_prepared_list = updated_list_for_or_operator
```
Comment (ababic) [inline etna/ciim/client.py:129]: Is it not possible to do this in the above loop? And can we not just always add `:or` whether multiple values have been supplied or not, or does that cause an error?

**Comment by TNA-Allan** [comment]
> Where there are _horrible API-specific implementation details_, 

_Search your feelings Luke_  :) 

> Perhaps we can add a `or_filter_fields` option or something to `KongClient.search()`, and that can pass both values on to the `prepare_filter_aggregations()`?

I understood previously -
> What I mean is: this behaviour doesn't belong on the client.

I thought lets keep it simple and leave the `prepare_filter_aggregations()` do what its doing.

File: etna/ciim/client.py (hunk 1 above)
Comment (TNA-Allan) [reply etna/ciim/client.py:129]: Don't want to "mash" adding `:or` whilst preparing the filter to remove spaces in the above loop. But as a role of preparing the filter, added after remove spaces, to keep it simple and target only those configured fields in var  `fields_using_or_operator`

**Review by ababic — APPROVED** [review]

**Pull request merged**